File: flutter_base_05/lib/modules/recall_game/game_logic/models/power_cards/base_power_card.dart
/// Base Power Card Class
///
/// This module defines the base class for all power cards in the Recall game.
/// Each power card extends this class and implements its own special logic.

abstract class BasePowerCard {
  /// Base class for all power cards
  
  final dynamic gameState;
  late final String cardName;

  BasePowerCard(this.gameState) {
    cardName = runtimeType.toString().toLowerCase();
  }

  Map<String, dynamic> updateDecks() {
    /// Update the draw and discard piles
    return {};
  }

  Map<String, dynamic> updateComputerPlayers() {
    /// Update computer player information
    /// purpose is to update the computer players' data like hand/known from other players/points etc etc by calling the player class methods
    return {};
  }
}

File: python_base_04/core/modules/recall_game/models/power_cards/base_power_card.py
"""
Base Power Card Class

This module defines the base class for all power cards in the Recall game.
Each power card extends this class and implements its own special logic.
"""

from abc import ABC, abstractmethod
from typing import Dict, Any, Optional
from ...game_logic.game_state import GameState
from ..player import Player


class BasePowerCard(ABC):
    """Base class for all power cards"""
    
    def __init__(self, game_state: GameState):
        self.game_state = game_state
        self.card_name = self.__class__.__name__.lower()

    def update_decks(self) -> Dict[str, Any]:
        """Update the draw and discard piles"""

    
    def update_computer_players(self) -> Dict[str, Any]:
        """Update computer player information"""
        # purpose is to update the computer players' data like hand/known from other players/points etc etc by calling the player class methods